{
  char[] x={0};
  boolean thrown=false;
  try {
    Array.setChar(x,0,(char)1);
  }
 catch (  Exception e) {
    fail("Exception during get test : " + e.getMessage());
  }
  assertEquals("Get returned incorrect value",1,Array.getChar(x,0));
  try {
    Array.setChar(new Object(),0,(char)9);
  }
 catch (  IllegalArgumentException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Passing non-array failed to throw exception");
  }
  thrown=false;
  try {
    Array.setChar(x,4,(char)9);
  }
 catch (  ArrayIndexOutOfBoundsException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Invalid index failed to throw exception");
  }
  thrown=false;
  try {
    Array.setChar(null,0,(char)0);
  }
 catch (  NullPointerException e) {
    thrown=true;
  }
  if (!thrown) {
    fail("Null argument failed to throw NPE");
  }
}
